### DISCUSSION TITLE
Falling tree config file not allowing whitelisting properly

### DISCUSSION THREAD

**Discussion by bronasr** [RakambdaOrg/FallingTree#538] [Q&A] [answered]
So i've been reading all 38 pages of comments on the curseforge page, as well as reading the discussions here on the github, but I can't seem to get the .cfg file to play nicely in terms of the item white/black listing. All of the discussions that show easy configuration methods appear to reference a .toml that is not present in the single player or server config folders. I think I must just be typing it in incorrectly, below is what the file looks like. if I replace the < with an = and type ["mod:itemid"] it gives no feedback, but doesn't allow that item to work. If I leave the karat < and type ["mod:itemid"] it actually marks the entire config file as errored and generates a new one. I've tried several different ways of entering the items that I'd like white/black listed, but it doesn't seem to be making a difference. For reference, I'm playing on 1.12.2 using forge. Falling tree is working as expected in all other ways. Any help would be greatly appreciated.

tools {
    # List of tools that should not be considered as tools.
    # INFO: This wins over the whitelist.
    S:blacklisted <
     >

    # Defines the number of times the damage is applied to the tool.
    # ie: if set to 1 then breaking 5 logs will give 5 damage.
    # ie: if set to 2 then breaking 5 logs will give 10 damage.
    # If set to 0, it'll still apply 1 damage for every cut.
    # INFO: This only applies when the tree is cut when using the mod.
    # Min: 0
    # Max: 2147483647
    I:damage_multiplicand=1

    # When set to true, the mod will be activated no matter what you have in your hand (or empty hand).
    # INFO: Blacklist still can be use to restrict some tools.
    B:ignore_tools=false

    # When set to true, when a tree is broken and the tool is about to break we will just break enough blocks so that the tool is left at 1 of durability.
    B:preserve=false

    # Applies a speed modifier when breaking the tree.
    # 0 will disable this, so the speed will be the default one of breaking a block.
    # If set to 1 each log block will be counted once, so if the tree is 5 blocks tall it'll require the time of breaking 5 logs.
    # If set to 2 each log block will be counted twice, so if the tree is 5 blocks tall, it'll require the time of breaking 10 logs
    # INFO: Only in INSTANTANEOUS mode.
    # WARNING: If you are on a server, this either has to be set to 0 or every player should have the mod. Else they'll have a weird effect of breaking the block but the block is still there.
    # Min: 0.0
    # Max: 50.0
    D:speed_multiplicand=0.0

    # Additional list of tools that can be used to chop down a tree.
    # INFO: Items marked with the axe tag will already be whitelisted.
    S:whitelisted <
     >
}

**Comment by Rakambda** [comment, marked as answer]
File format changed in some version (I don't remember which). In newer versions it's JSON format so the format you saw `"key": ["val1", "val2"]`. 

Now 1.12 uses old format, which is the Forge TOML format (and is weird).
If I remember well you need something like : 
```
# Additional list of tools that can be used to chop down a tree.
# INFO: Items marked with the axe tag will already be whitelisted.
S:whitelisted <
    mod_id1:id1
    mod_id2:id2
    mod_id3:id3
 >
```

**Reply by bronasr** [reply]
Thanks so much I was working on that last night for no joke 4 hours. It looks like that method of listing worked. I created a github account just so I could ask the question.
